walkeeper/src/pq_protocol.rs

```diff
@@ -81,7 +74,11 @@ impl FeStartupMessage {
             _ => StartupRequestCode::Normal,
         };
 
-        let params_bytes = &buf[8..len];
+        // FIXME: A buffer pool would be nice, to avoid zeroing the buffer.
```

Comment:
It's cheap enough that it won't matter. If there was a `read_exact` function created the vector for us, or if the compiler could elide the zeroing, that would be cool, but it's not really a performance issue here.

Maybe if we had a buffer that we reuse for the other messages anyway, we could use that here too without any extra code, but that'd more for the sake of consistency than performance.

Reply:
I think when I wrote that comment, I had a lot more temporary buffers. But a lot of them went away as I refactored more code to deserialize directly from the `Read` interface.

Since there aren't very many temporary buffers left, and the compiler will probably eliminate most of the zeroing, I removed the comment.